### DISCUSSION TITLE
Is this an "Invalid Address Validation in Ownership Transfer" Vulnerability? (does it allow ownership transfer to a zero address?)

### DISCUSSION THREAD

**Discussion by Steiner-254** [Cyfrin/security-and-auditing-full-course-s23#245] [Q&A]
Hi team,

So for a while thought this is an issue - "Invalid Address Validation in Ownership Transfer - Allows Zero Addresses" Am I right???

```solidity
 function setContractOwner(address _newOwner) internal {
        GoldStorage storage gs = goldStorage();
        address previousOwner = gs.contractOwner;
        gs.contractOwner = _newOwner;
        emit OwnershipTransferred(previousOwner, _newOwner);
}
 ```

Lemme know your thoughts ;p

**Comment by EngrPips** [comment]
Hello @Steiner-254, Seem like this discussion is out of the purpose scope of this Github but if that function is meant to be called by a trusted party then your findings will be invalid plus many static analyser would catch that already.
